File: 백준/Silver/10845. 큐/큐.go
package main

import (
	"bufio"
	"os"
	"strconv"
)

var (
	sc  *bufio.Scanner
	bw  *bufio.Writer
	que []int
)

func init() {
	sc = bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	bw = bufio.NewWriter(os.Stdout)
}

func scanInt() (n int) {
	sc.Scan()
	n, _ = strconv.Atoi(sc.Text())
	return
}

func main() {
	defer bw.Flush()
	N := scanInt()

	for i := 0; i < N; i++ {
		sc.Scan()
		order := sc.Text()
		switch order {
		case "push":
			num := scanInt()
			que = append(que, num)
		case "pop":
			if len(que) == 0 {
				bw.WriteString("-1\n")
			} else {
				out := que[0]
				que = que[1:]
				bw.WriteString(strconv.Itoa(out) + "\n")
			}
		case "size":
			bw.WriteString(strconv.Itoa(len(que)) + "\n")
		case "empty":
			if len(que) == 0 {
				bw.WriteString("1\n")
			} else {
				bw.WriteString("0\n")
			}
		case "front":
			if len(que) == 0 {
				bw.WriteString("-1\n")
			} else {
				bw.WriteString(strconv.Itoa(que[0]) + "\n")
			}
		case "back":
			if len(que) == 0 {
				bw.WriteString("-1\n")
			} else {
				bw.WriteString(strconv.Itoa(que[len(que)-1]) + "\n")
			}
		}
	}
}

File: 백준/Silver/10845. 큐/큐.cc
#include "bits/stdc++.h"

using namespace std;

int main() {
    queue<int> Q;

    int N;
    cin >> N;
    for (int i = 0; i < N; ++i) {
        string cmd;
        cin >> cmd;

        if (cmd == "push") {
            int num;
            cin >> num;
            Q.push(num);
        } else if (cmd == "pop") {
            if (Q.empty()) {
                cout << -1 << "\n";
            } else {
                cout << Q.front() << "\n";
                Q.pop();
            }
        } else if (cmd == "size") {
            cout << Q.size() << "\n";
        } else if (cmd == "empty") {
            if (Q.empty()) {
                cout << 1 << "\n";
            } else {
                cout << 0 << "\n";
            }
        } else if (cmd == "front") {
            if (Q.empty()) {
                cout << -1 << "\n";
            } else {
                cout << Q.front() << "\n";
            }
        } else if (cmd == "back") {
            if (Q.empty()) {
                cout << -1 << "\n";
            } else {
                cout << Q.back() << "\n";
            }
        }
    }

    return 0;
}